internal/pkg/template/workload.go

```diff
@@ -142,6 +140,14 @@ type SidecarOpts struct {
 	EntryPoint   []string
 	Command      []string
 	HealthCheck  *ContainerHealthCheck
+	PortMappings []*PortMapping
+}
+
+// PortMapping holds container port mapping configuration.
+type PortMapping struct {
+	Protocol      string
+	ContainerPort uint16
```

Comment:
This might be my illusion but it seems like we are converting  `uint16`, `uint64` and `string`  a lot. If so can we simplify this?

Reply:
You are right, but I have kept these values such that we will need to do less conversion while converting exposedPorts to PortMappings. Even if I change it ContainerPort to string we still need to make some conversions at other locations. Either ways conversions are required hence I at least tried making sure that  PortMapping and ExposedPort struct are similar.